### ISSUE TITLE
Publish to store

### ISSUE THREAD

**Issue by FrostByte2** [jellyfin/jellyfin-webos#28]
0.2.2 i have already made a ipk attatched in case dont want to make one.

it is inside the zip
[org.jellyfin.webos_0.2.2_all.zip](https://github.com/jellyfin/jellyfin-webos/files/4941854/org.jellyfin.webos_0.2.2_all.zip)

**Comment by anthonylavado** [comment]
@ghostrobo282 Please don't do this. You would be intentionally holding back our progress.

**Comment by Tyri0n** [comment]
I can see where he's coming from - having Jellyfin disappear from TV if you forget to periodically extend develop mode session is pretty irritating.

**Comment by anthonylavado** [comment]
@Tyri0n I understand, but trying to do things like this, even threatening to do so, has several effects:
- it could block our release with LG
- it could block our release on other platforms
- it puts a dampener on the motivation of contributors to this repository

A lot of people don't know the work that has to be put in behind the scenes to get something published, tested, and accepted for app stores and vendor requirements. I put in several hours of work for each place we offer our apps. I can safely say the TV app stores are by far the most restrictive, requiring the most work of all to go through stacks of agreements, documents, many e-mail messages, and more. I'd rather get a root canal in some cases. Publishing on the Apple App Store is a cakewalk by comparison.

We don't immediately submit the apps because these places have stringent requirements (as they should) to ensure that an app works the way a user expects it to when they download it to their TV. There's years worth of compatibility to optimize for, and that's still a work in progress in the main web repo (which is what this serves).

I want the many hours of work that our contributors have done (for free and in their spare time!) to be respected properly, not dismissed because anyone can't wait for it to be just right.

This issue opener even noted in another comment of theirs (and in a YouTube video they posted) that it doesn't work correctly in all cases.

**Comment by Tyri0n** [comment]
@anthonylavado Fair enough and thanks for your efforts. Any idea on when it's likely to be available on the app store?

**Comment by anthonylavado** [comment]
@Tyri0n For that, I will defer to our contributors: https://forum.jellyfin.org/t/jellyfin-for-lg-webos/236/23?u=anthonylavado

A large portion of the app will only really work with 10.6, so it will have to be at least after that gets released.

**Comment by Tyri0n** [comment]
@anthonylavado Lovely, thanks. Will keep a keen eye on that thread. Keep up the good work!

**Comment by ghostrobo282** [comment]
I hope to see more progress soon thanks for the update on the app I have noticed the bug with the tv shows tab mine on my TV puts the words tvshow on top of the shows tab on top anyone else having this issue or know how to fix it would be greatly appreciated hope to see the project expand

**Comment by Rohaq** [comment]
@anthonylavado Is there a tracked checklist for what needs to be done to get the app into a good enough state for publishing?

It might encourage more contributions if people had a list of requirements to burn through if they're motivated enough to get this on the store! 🙂

**Comment by MrTimscampi** [comment]
@Rohaq The big thing is going through [this document](http://webostv.developer.lge.com/download_file/pforce/9378/2147/), checking everything, and fixing what doesn't pass LG's QA.

Pretty much all of the fixes will happen on https://github.com/jellyfin/jellyfin-web/ and not here, though.

**Comment by edrevo** [comment]
I am getting "Invalid File" with the link provided by @MrTimscampi. Do I need to login somewhere to be able to see the file or is the link invalid?

**Comment by dkanada** [comment]
It turns out you do need to login for the link to work.

**Comment by ghostrobo282** [comment]
There is a link to my video

**Comment by surdu** [comment]
> It turns out you do need to login for the link to work.

Login where? There is no login link / box.

@anthonylavado Other than going through that checklist, is there anything else that the community can help with to make this happen? Hacktoberfest is coming, and might be a nice idea to mark with the "hacktoberfest" label some issues that can push this publishing along.

**Comment by anthonylavado** [comment]
@surdu That is a good idea. If I'm not mistaken though, a lot of the work has to take place in Jellyfin/Jellyfin-web, which is undergoing a bug change.

**Comment by surdu** [comment]
@anthonylavado Then in addition to the "hacktoberfest" label, the issues that help publishing the app on WebOS could be labeled as such in the jellyfin-web project.

**Comment by Brice187** [comment]
Any ETA fir this one? Or is it possible to extend developer mode for more than 50h?

**Comment by Rohaq** [comment]
> Any ETA fir this one? Or is it possible to extend developer mode for more than 50h?

Developer mode is a restriction on the LG itself - there's nothing the app can do to get around it, aside from being officially published on the LG Store.

**Comment by oddstr13** [comment]
Looks like the mentioned checklist download isn't hot-linkable.
Clicking the `documents_for_app_qa.zip` link on this page works; http://webostv.developer.lge.com/design/ux-checklist/

**Comment by cj2tech** [comment]
Any update on this or eta? Legit been waiting forever for this. Been the last price for me before I can't fully switch over to jellyfin lol.

**Comment by Brice187** [comment]
Would love to see a list of tasks which should be done to get the app published in the LG store. Maybe I could work on some of those tasks.

The app is published on other major platforms like AndroidTV and FireTV. I think the QA requirements are nearly the same for webOS...

**Comment by MrTimscampi** [comment]
> The app is published on other major platforms like AndroidTV and FireTV

These are a completely different app, made specifically for Android. This one essentially runs the web client.

The best way to help get this published would be to go over LG's QA checklist and fix the items that don't meet their requirements. The checklist is linked above in @oddstr13 's post.

**Comment by Brice187** [comment]
Okay, so here is the QA Checklist https://docs.google.com/spreadsheets/d/1505AA1OYBr64qHIYbQGkd0eyevDI1iUhtr8HK9ZbpKM/edit?usp=sharing 

I will be working on some the items, contributions are welcome. When all points are filled, i will hand it over to the project maintainers and hope the will submit the app soon :)

**Comment by Brice187** [comment]
Filled out pretty much of the forms, but I would love to have some jellyfin internal QA :-). Especially some "developer comments" would be nice

Who is in charge for submitting the app to the store? Some task can not be prefilled, because they have to be done on the fly.

**Comment by Brice187** [comment]
@anthonylavado hey, i filled out the needed forms:

# self evaluation
[self_evaluation_checklist_4.0_jellyfin.xlsx](https://github.com/jellyfin/jellyfin-webos/files/5394404/self_evaluation_checklist_4.0_jellyfin.xlsx)

Please take a closer look at items 1, 3, 49

# ux scenario
[ux_scenario_document_4.3_jellyfin.pptx](https://github.com/jellyfin/jellyfin-webos/files/5394402/ux_scenario_document_4.3_jellyfin.pptx)

Would be great to get some feedback. Please tell me if you need more information. Maybe we can get jellyfin submitted to the LG store in this year :)

**Comment by Brice187** [comment]
@anthonylavado hey, i filled out the needed forms:

# self evaluation
[self_evaluation_checklist_4.0_jellyfin.xlsx](https://github.com/jellyfin/jellyfin-webos/files/5394434/self_evaluation_checklist_4.0_jellyfin.xlsx)

Please take a closer look at items 1, 3, 49

# ux scenario
[ux_scenario_document_4.3_jellyfin.pptx](https://github.com/jellyfin/jellyfin-webos/files/5394402/ux_scenario_document_4.3_jellyfin.pptx)

Would be great to get some feedback. Please tell me if you need more information. Maybe we can get jellyfin submitted to the LG store in this year :)

**Comment by anthonylavado** [comment]
I'll go over this. Thanks!

**Comment by Brice187** [comment]
Hey @anthonylavado, 

Did you get the time to take a closer look ? :)

**Comment by EraYaN** [comment]
@Brice187 The pttx should only contain licensed media in screenshots, so we'll have to use the demo server for screenshots (and probably the login bit as well) Other than that I really like it and I hope LG likes it to.

**Comment by MrTimscampi** [comment]
> cause then we need to go and hide the dashboard on LG TVs.

That's done in 10.7 already :wink:

**Comment by Brice187** [comment]
@EraYaN Removed licensed media and used public domain / Creative Commons Attribution 3.0 (https://peach.blender.org/about/) media.

I do not have a 1drv account, so i uploaded the file here:
[ux_scenario_document_4.3_jellyfin.pptx](https://github.com/jellyfin/jellyfin-webos/files/5505476/ux_scenario_document_4.3_jellyfin.pptx)

**Comment by EraYaN** [comment]
Thank you for the changes, I have updated the hosted document.

**Comment by Bolkarr** [comment]
Quick question, since it is not possible to watch any 4k HEVC videos on any browser, will they work if the WebOS app is a wrapper for the web development?

**Comment by Nickbert7** [comment]
Is there any uptate on getting the app published to the store?

**Comment by mastha7** [comment]
Hi,

@anthonylavado @EraYaN any updates? How we can help?

**Comment by dkanada** [comment]
I'm not sure on the current status, but right now I think the team is waiting on 10.8.0 for some better support for older browsers. I can't say exactly since I'm not involved in the WebOS client, but I'm not aware of any blockers. Maybe the experience on 10.7.0 is good enough for a release.

**Comment by XanderStrike** [comment]
10.7.0 is absolutely good enough for release. The only issue I have with it is having to reinstall every 50 hours. I wonder if we can get @anthonylavado to weigh in since they're the one who initially wanted to hold it back?

**Comment by anthonylavado** [comment]
@XanderStrike I wasn't _deliberately_ holding anything back, I was waiting for the web team to tell me "Okay, this is good to go now". My understanding is that this is the case with 10.7, so we have already been preparing on the backend for it.

**Comment by dovecode** [comment]
I just found this thread. Is there anything a non-WebOS developer can do to help progress this (review docs, test builds, etc), or should I take @anthonylavado's message abovce to mean that it is all done and just a matter of time?

**Comment by da-anda** [comment]
I used 10.7 webinterface version on my LG and it works good enough, even purely via the browser on the TV. The only annoyance is that it won't stay fullscreen when you hit back on the remote. So looking forward to the app.

**Comment by Brice187** [comment]
Thank you for keeping us updated 😎

**Comment by supersebbo** [comment]
@anthonylavado Extrapolating from your comments on Jan 2, is the WebOS app going to get published in the LG store along side the JF 10.7 release?  If not - repeat the ask if there is anything the community can do to help?  Has the submission been review by LG yet and is there any specific feedback that needs action?  Thanks!

**Comment by bronson** [comment]
Just installed it and yall are right: the web interface works great. Much better than I'd expect. I almost don't believe my eyes and ears. Great work.

One other annoyance is scrolling. Because the autoscroll boundaries are so large and the autoscroll so rapid, it's frustrating to try to click something in the middle of a list. Just when you're getting close, it jumps away. Far away. The scroll wheel works well though.

**Comment by iMartyn** [comment]
Agreed, I found that a "magic wand" style remote was kinda impossible to use, but the "keyboard" arrow controls work fine. (I'm on an earlier model that doesn't have a magic remote, mine is an aftermarket that registers as mouse+kb).  That being said, I doubt LG is going to go past the login screen anyway because they're probably not going to install a jellyfin server to test the app.

**Comment by ferferga** [comment]
@iMartyn We provide to stores reviewers information about how to use our demo servers. That's why we pass most of the reviewing process enforced by them.

**Comment by DpunktKpunkt** [comment]
Now that 10.7 is released I can't hardly wait finding Jellyfin in LG Store :)

**Comment by anthonylavado** [comment]
Well, LG has advised all developers that there is a backlog causing reviews to take up to one month...

**Comment by celerydoor** [comment]
@anthonylavado Does that mean it's been submitted to LG and we're now waiting on their review?

**Comment by Tyri0n** [comment]
w00t!

**Comment by anthonylavado** [comment]
Rejected again by "pre-QA". This one took longer than just a day, so it seems they might have actually tried.

Off to read their comments and go for round three...

**Comment by sorano** [comment]
Shame, I mean that webos tv store is filled with 95% crap apps so it's weird that they reject a quality project like this (for probably bullshit reasons).

I hope the constant rejecting dont wear your morale down.

Best of luck!

**Comment by DpunktKpunkt** [comment]
Bummer! Are you going to publish their comments for better testing / fixing with more people?

**Comment by anthonylavado** [comment]
> Bummer! Are you going to publish their comments for better testing / fixing with more people?

@DpunktKpunkt So far it's nothing that anyone here can action. Thanks for the offer to help though!

**Comment by celerydoor** [comment]
Could you share the full response by LG anyway? I think there's a lot of frustration in this segment of the community with the multi-year effort to get published in the WebOS store. Transparency with the process would help a lot.

**Comment by anthonylavado** [comment]
The icons needed to be set to transparent, and version number incremented which was fixed in the first round, and this one is cantered with access to our private demo server for testing purposes. There's not much else to report.

We've filled in the documents and the paperwork, I've taken the screenshots, and performed tons of validation testing with the emulator on different versions.

Believe me, I'm trying to get it on there. I understand you want it - I don't even have an LG TV and I want to get it out for you. I know you didn't mean it this way, but it sounded like a demand, so I initially didn't want to share. Heck, I didn't even want to confirm that I had submitted it out of fear…

Anyway, I'll keep you updated as I can. I'll stop short of sharing their exact messages because they'd probably frown on that. All this testing happens in Korea, so the timelines can be weird (getting emails at 4 am my time).

**Comment by supersebbo** [comment]
Thanks for sharing @anthonylavado! Sounds like relatively minor issues, you're getting closer every day!  We appreciate your efforts, and as always let us know if there is anything we can help with.

**Comment by anthonylavado** [comment]
We've figured out our networking issues and I've streamlined some of the server config to make LG's Testing as easy as possible. Hopefully this works.

**Comment by qwerty2k** [comment]
Don’t have anything constructive to add other than thank you @anthonylavado for your work on this especially as you don’t even have an LG tv. Much appreciate by all.

**Comment by anthonylavado** [comment]
Not rejected yet, but still in "QA Processing". Waiting.

**Comment by iqlogic** [comment]
I've just bought 55" LG GX and already tried web app and it is working surprisingly well. I didn't expect that built-in browser can handle such a thing 😀 
I looking forward to native app, your work is really astonishing 🤩
Huge thanks for your effort and passion 😎

**Comment by CalvoUTN** [comment]
Thanks for the update! I'm just getting started with Jellyfin after ditching Plex for a myriad of issues. I'm using it in the web app and except scrolling it works perfectly. Thanks for all the work!

**Comment by AlexKalopsia** [comment]
Thank you for the hard work @anthonylavado <3

**Comment by gtrias** [comment]
Thanks a lot to address this issue @anthonylavado . Looking forward to have it installed permanently :)

**Comment by mastha7** [comment]
Hello. 
Any news? Still in QA process?

**Comment by dkanada** [comment]
They're still testing the client internally.

**Comment by elvis0288** [comment]
hello guys, i am following this to see when it gets publish but i did compiled the app and put it on my LG TV. now i am not sure if this is something that will cause the app to be reject again or if this is something specific to me but the resume button is not working at all.. that is the only problem i can see after testing for more than a week

**Comment by TimMods** [comment]
@elvis0288 don't worry. What you're doing on your TV doesn't concern here :)

Very much looking forward to the store app. One of the things I'll be missing is launching Jellyfin by the 'MOVIES' button on the LG remote control. I've change the packagename to "ui30", which is the packagename of RakutenTV. The button launches Jellyfin directly now instead.

**Comment by anthonylavado** [comment]
Rejected, but we actually got through the testing process this time. This means that we have a spreadsheet with video examples of what's not working, and what needs to be fixed (the first two rejections had to do with metadata and server access).

I will be closing a few existing issues on the board to make room for the project/issues to resolve what's needed for this to get released. We have two more attempts to get this right, after which we get restricted from submission for three months.

**Comment by k-matti** [comment]
Please tell us what is not working and needs to be fixed?

**Comment by anthonylavado** [comment]
@k-matti That's what I'm going to be doing, please wait 🙂

**Comment by chrisaut** [comment]
Just want to say thank you to everyone working on this, especially those who don't even have an LG TV.

I also don't understand LG with this, it's not like they have a ton of Apps in the store, you would think they would be a bit forgiving and open to get more apps in there.

**Comment by anthonylavado** [comment]
The list of issues is now up. Here's a link that filters just to them: https://github.com/jellyfin/jellyfin-webos/issues?q=label%3A%22release+critical%22+

I'd appreciate any help that users in here can provide. I've tried to include as much information as possible. If you have any questions, please ask in the specific issue, so we can keep the discussion organized.

**Comment by Dycelll** [comment]
Awesome, thank you! I'm no programmer but I'll try to validate the issues and help with troubleshooting!

**Comment by illspirit** [comment]
> 
> 
> The list of issues is now up. Here's a link that filters just to them: https://github.com/jellyfin/jellyfin-webos/issues?q=label%3A%22release+critical%22+
> 
> I'd appreciate any help that users in here can provide. I've tried to include as much information as possible. If you have any questions, please ask in the specific issue, so we can keep the discussion organized.

Just bought an LG TV - thank you for doing this.

**Comment by anthonylavado** [comment]
Just popping in to say we're still working on this, piece by piece. Going to submit soon, just have to get the resume and playback issues worked out. We updated the server recently so we're testing against those fixes.

**Comment by andreapl** [comment]
Many thanks for your effort !!!

**Comment by briansr938** [comment]
> Just popping in to say we're still working on this, piece by piece. Going to submit soon, just have to get the resume and playback issues worked out. We updated the server recently so we're testing against those fixes.

Thanks for the update.

**Comment by dgalli1** [comment]
For the love of god can you all stop pinging every person in this thread, just to say how greatful you are. Reactions are implemented for a reason.

**Comment by AlexKalopsia** [comment]
Ok



(i am so sorry)

**Comment by knockhart** [comment]
Hi there!

Is there any update on the project? I've installed it to my LG webos, and on local network it works fine. After a while the dev mode deletes jellyfin. Can't wait to get the app from LG content center. 
Thanks for your all efforts making this possible!

**Comment by anthonylavado** [comment]
Locking to avoid new comments. Use reactions. I'll update you when I can.

**Comment by anthonylavado** [comment]
For those interested in it, a new IPK has been uploaded to the GitHub release page for this repository. Submission to LG is still forthcoming, but the channel is available through a homebrew channel, if you were interested in rooting the TV. Details are available here: https://github.com/jellyfin/jellyfin-webos/issues/67

For those that still want to keep this installed directly with Dev Mode, there are lots of ways to make sure Dev Mode stays renewed and doesn't expire in the 48 hour window. A Reddit post with details is here: https://www.reddit.com/r/jellyfin/comments/ryowwb/i_created_a_simple_script_to_renew_the_devmode_on/

For future updates, we did merge a fix regarding DTS playback not being supported in 2020/2021+ models of LG TVs. Starting with Jellyfin 10.8 (currently in Alpha), DTS audio will be transcoded to AC-3.

This issue has more than 40 people attached, so this will be my final update here before closing it to start a new issue for tracking. I've got the emulator open, so let's get testing.
